desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -293,6 +243,11 @@ public void onRemove(final InputColumn<?> sourceColumn) {
     private static final int DEFAULT_WINDOW_WIDTH = 1000;
     private static final int DEFAULT_WINDOW_HEIGHT = 710;
 
+    private static final int SERVER_CHECK_INTERVAL = 2000; // [ms]
+    private DescriptorProviderState _descriptorErrorState;
+    private Thread serverCheckingThread;
```

Comment:
It looks like the thread is completely controlled by the ServerChecker class. So I would suggest not to save the thread as a field in this class - this class is already very big and should not increase in complexity. Avoiding the extra field helps in that.